### PR TITLE
chore: add custom AI agents and layered CLAUDE.md guides for AI-driven development

### PR DESCRIPTION
**Pull request by rvignesh89**
## Summary

This PR establishes the foundation for **AI-driven development** of Glific by adding three specialized Claude subagents and layered source-of-truth documentation in key directories.

### Agents (`.claude/agents/`)

| Agent | Role | Key capabilities |
|-------|------|-----------------|
| `backend-engineer` | Primary implementer | Full vertical slice (migration → schema → context → GraphQL → resolver → schema wiring → .gql assets → Bruno docs); multi-tenancy invariants; module scope enforcement |
| `test-automator` | Test engineer | API-first test strategy (ConnCase default, DataCase for edge cases); GraphQL fixtures + .gql assets; external mocks; coverage |
| `code-reviewer` | Quality gate | 6 priority review dimensions: multi-tenancy/security, GraphQL completeness, module scope/API design, correctness, data layer, tests |

Key conventions encoded in the agents:
- **Bruno API docs are a contract artifact** — every new query/mutation gets a `api.docs/` entry
- **API-first tests** — ConnCase (GraphQL boundary) by default; DataCase only for sub-API logic
- **Module scope enforcement** — flags large unfocused modules, UI-coupled field names, dumping-ground additions
- **Strict Credo** (`mix credo --strict`) — matches actual CI configuration in `.check.exs`

### Layered CLAUDE.md files

| File | Source-of-truth for |
|------|---------------------|
| `lib/glific/CLAUDE.md` | Contexts, schemas, Oban workers, multi-tenancy invariant, error handling, caching |
| `lib/glific_web/CLAUDE.md` | GraphQL types/resolvers, schema.ex wiring, authorization, .gql assets |
| `test/CLAUDE.md` | Case template selection, DataCase/ConnCase shapes, fixtures, mocking, coverage |
| `priv/repo/migrations/CLAUDE.md` | Migration conventions, hard rules, safe operations on large tables |

### Bug fixes

Fixes two pre-existing compiler warnings that were blocking the `mix check` pre-commit hook:
- `lib/glific/partners.ex:1054` — unused default argument `organization_id \\ nil`
- `lib/glific/third_party/bigquery/bigquery.ex:631` — unused default argument `organization_id \\ nil`

## Test plan

- [x] `mix check` passes (compiler ✓, credo ✓, dialyzer ✓, doctor ✓, mix_format ✓)
- [x] All files are documentation/configuration only — no runtime behavior changed (except the two warning fixes)
- [x] The two warning fixes remove unused default arguments from `defp` functions that were always called with all arguments

🤖 Generated with [Claude Code](https://claude.ai/claude-code)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,11 +1,13 @@
 # The directory Mix will write compiled artifacts to.
 /_build/
+/_build
```
Comment (rvignesh89) [inline .gitignore:3]: To avoid symlinks be staged

File: test/glific/templates_test.exs [hunk 2]
```diff
@@ -179,35 +179,23 @@ defmodule Glific.TemplatesTest do
 
       assert length(session_template_list) == 3
 
-      # Match term with label of associated tag
-      template = session_template_fixture(Map.merge(attrs, %{label: "label4"}))
+      # Match term with label of associated tag (via direct tag_id FK on session_template)
```
Comment (rvignesh89) [inline test/glific/templates_test.exs:182]: Fixes flaky tests

**Review by AmishaBisht — APPROVED** [review]
